README.md

```diff
@@ -1,130 +1,182 @@
+
+First, if you use Smartcar hosting you will add a redirect URI to your application of the form `https://javascript-sdk.smartcar.com/redirect?app_origin=<your-client-uri-here>`. So for example, if your application URI was `https://my-awesome-app.com` you'd add the following redirect URI to your app in the dashboard: `https://javascript-sdk.smartcar.com/redirect?app_origin=https://my-awesome-app.com`. Note, this value is only your app's origin, if your application was at `https://my-awesome-app.com/application` you would still have the example redirect URI listed without the `/application`. Any restrictions on normal redirect URIs (like requiring `https` when not using `localhost` still apply to the value you give for `app_origin`). See the [Register](https://smartcar.com/docs#register) section of our docs for valid formats.
+
+Second, if you use Smartcar hosting you must provide an `onComplete` function with the following signature `onComplete(err, code, state <optional>)` to the Smartcar constructor. This `onComplete` method will be called when the user completes the authorization flow. If they approved access you'll need to exchange the `code` parameter for an access token, otherwise you'll need to handle the given `err` parameter holding the error.
+
+If you host the redirect yourself you can provide any valid redirect URI (see the [Register](https://smartcar.com/docs#register) section of our docs) and `onComplete` is optional. You may source `redirect.js` in your redirect popup to close out the window and trigger firing of the `onComplete` (if you've provided one to the `Smartcar` constructor).
+
+Here are example flows for each method:
+
+### Example Flows
+
+#### Smartcar Hosted
+
+1. User clicks "Connect your car" button (or similar) on your application's website.
+2. The user is redirected to an OAuth authentication page, using the `openDialog` or `addClickHandler` methods. This page requires the user to authenticate with their vehicle credentials. **Note**: because the Smartcar hosted redirect page uses `postMessage` to communicate the access code back to the client page the redirect page must be opened in a separate window.
+3. After entering credentials the user will be presented with a list of their vehicles to authorize and a selection of permissions to allow your app.
+4. Once they've selected their vehicles & permissions Smartcar's services will redirect to the Smartcar hosted redirect URI. This page will load `redirect.js` which will post a message to the original client page that opened the redirect. The `Smartcar` class instantiated on the client page handles this redirect, calling the required `onComplete` method provided to the constructor. The `onComplete` method you provided must handle an error if the user refused authorization or sending the code to your backend to exchange for an access token if the user authorized your application.
```

Comment:
Can rephrase the first couple sentences? It seems kind of hard to understand and some terms here seem quite ambiguous.